Issue with package local dist into pex

**Describe the bug**
When I attempt to build a pex that depends on a local dist, it fails when PEX is creating a symlink (see logs below).
( Using this feature: #13306 )

Verified with a simpler case in the example repo that what I try to do works.

**Pants version**
2.8.0rc5 (also observed on latest `main`)

**OS**
MacOS

**Additional info**

This is the process and resulting exception from the debug logs:


```
11:52:03.82 [DEBUG] Running Extracting 12 requirements to build demo.pex from repository.pex: PyYAML==5.3.1, backoff==1.10.0, blinker==1.4, click==7.1.2, codetiming==1.2.0, enum34==1.1.10, gevent==20.9.0, marshmallow==3.9.0, packaging==20.9, pluggy==0.13.1, prometheus-client==0.9.0, transitions==0.8.4 under semaphore with concurrency id: 12
11:52:03.85 [DEBUG] spawned local process as Some(82860) for Process { argv: ["/Users/x/.pyenv/versions/3.7.12/bin/python", "./pex", "--cert", ".cacert.pem", "--python-path", "/Users/x/.pyenv/versions/3.7.12/bin:/Users/x/.pyenv/versions/3.8.11/bin:/Users/x/.pyenv/versions/3.9.7/bin:/usr/local/bin:/root/pyenv/bin", "--tmpdir", ".tmp", "--output-file", "demo.pex", "--requirements-pex", "local_dists.pex", "--pex-repository", "repository.pex", "--interpreter-constraint", "CPython<4,>=3.8", "--no-emit-warnings", "--jobs", "6", "--manylinux", "manylinux2014", "--entry-point", "bazooka.cli:main", "--sources-directory=source_files", "--constraints", "constraints.txt", "PyYAML==5.3.1", "backoff==1.10.0", "blinker==1.4", "click==7.1.2", "codetiming==1.2.0", "enum34==1.1.10", "gevent==20.9.0", "marshmallow==3.9.0", "packaging==20.9", "pluggy==0.13.1", "prometheus-client==0.9.0", "transitions==0.8.4"], env: {"CPPFLAGS": "", "LANG": "en_US.UTF-8", "LDFLAGS": "", "PATH": "/Users/x/.cache/pants/pants_dev_deps/Darwin.x86_64.i386.py38.venv/bin:/Users/x/src/svs/python/bazooka/.dev-env/bin:/Users/x/.cargo/bin:/Users/x/.nvm/versions/node/v14.15.1/bin:/Users/x/.pyenv/shims:/Users/x/.pyenv/bin:/Users/x/Library/Python/3.7/bin:/usr/local/opt/ruby/bin:/usr/local/lib/ruby/gems/2.7.0/bin:/usr/local/opt/gettext/bin:/usr/local/opt/curl/bin:/Users/x/bin:/Users/x/src/svs/go/bin/:/Users/x/.jenv/shims:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/puppetlabs/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Applications/Wireshark.app/Contents/MacOS", "PEX_IGNORE_RCFILES": "true", "PEX_PYTHON_PATH": "/Users/x/.pyenv/versions/3.7.12/bin:/Users/x/.pyenv/versions/3.8.11/bin:/Users/x/.pyenv/versions/3.9.7/bin:/usr/local/bin:/root/pyenv/bin", "PEX_ROOT": ".cache/pex_root"}, working_directory: None, input_files: Digest { hash: Fingerprint<5bf6d09f5c6be72dc85edcf960884054b444c8d2cf8f8cab320260704a915adf>, size_bytes: 604 }, output_files: {RelativePath("demo.pex")}, output_directories: {}, timeout: None, execution_slot_variable: None, description: "Extracting 12 requirements to build demo.pex from repository.pex: PyYAML==5.3.1, backoff==1.10.0, blinker==1.4, click==7.1.2, codetiming==1.2.0, enum34==1.1.10, gevent==20.9.0, marshmallow==3.9.0, packaging==20.9, pluggy==0.13.1, prometheus-client==0.9.0, transitions==0.8.4", level: Info, append_only_caches: {CacheName("pex_root"): CacheDest(".cache/pex_root")}, jdk_home: None, platform_constraint: Some(Macos_x86_64), use_nailgun: Digest { hash: Fingerprint<e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855>, size_bytes: 0 }, cache_scope: Successful }

[...]

11:52:05.97 [ERROR] 1 Exception encountered:

  ProcessExecutionFailure: Process 'Extracting 12 requirements to build demo.pex from repository.pex: PyYAML==5.3.1, backoff==1.10.0, blinker==1.4, click==7.1.2, codetiming==1.2.0, enum34==1.1.10, gevent==20.9.0, marshmallow==3.9.0, packaging==20.9, pluggy==0.13.1, prometheus-client==0.9.0, transitions==0.8.4' failed with exit code 1.
stdout:

stderr:
Traceback (most recent call last):
  File "/Users/x/.cache/pants/named_caches/pex_root/unzipped_pexes/01a8d22400b26b1d132a9849fb6e28962f1ecf9e/.bootstrap/pex/pex.py", line 476, in execute
    exit_value = self._wrap_coverage(self._wrap_profiling, self._execute)
  File "/Users/x/.cache/pants/named_caches/pex_root/unzipped_pexes/01a8d22400b26b1d132a9849fb6e28962f1ecf9e/.bootstrap/pex/pex.py", line 401, in _wrap_coverage
    return runner(*args)
  File "/Users/x/.cache/pants/named_caches/pex_root/unzipped_pexes/01a8d22400b26b1d132a9849fb6e28962f1ecf9e/.bootstrap/pex/pex.py", line 432, in _wrap_profiling
    return runner(*args)
  File "/Users/x/.cache/pants/named_caches/pex_root/unzipped_pexes/01a8d22400b26b1d132a9849fb6e28962f1ecf9e/.bootstrap/pex/pex.py", line 532, in _execute
    return self.execute_entry(self._pex_info.entry_point)
  File "/Users/x/.cache/pants/named_caches/pex_root/unzipped_pexes/01a8d22400b26b1d132a9849fb6e28962f1ecf9e/.bootstrap/pex/pex.py", line 668, in execute_entry
    return self.execute_pkg_resources(entry_point)
  File "/Users/x/.cache/pants/named_caches/pex_root/unzipped_pexes/01a8d22400b26b1d132a9849fb6e28962f1ecf9e/.bootstrap/pex/pex.py", line 700, in execute_pkg_resources
    return runner()
  File "/Users/x/.cache/pants/named_caches/pex_root/installed_wheels/7fea57708be4117c2d962e01abd1585a8846dc06/pex-2.1.54-py2.py3-none-any.whl/pex/bin/pex.py", line 672, in main
    env=env,
  File "/Users/x/.cache/pants/named_caches/pex_root/installed_wheels/7fea57708be4117c2d962e01abd1585a8846dc06/pex-2.1.54-py2.py3-none-any.whl/pex/bin/pex.py", line 692, in do_main
    cache=ENV.PEX_ROOT,
  File "/Users/x/.cache/pants/named_caches/pex_root/installed_wheels/7fea57708be4117c2d962e01abd1585a8846dc06/pex-2.1.54-py2.py3-none-any.whl/pex/bin/pex.py", line 600, in build_pex
    pex_builder.add_distribution(installed_dist.distribution)
  File "/Users/x/.cache/pants/named_caches/pex_root/installed_wheels/7fea57708be4117c2d962e01abd1585a8846dc06/pex-2.1.54-py2.py3-none-any.whl/pex/pex_builder.py", line 458, in add_distribution
    dist_hash = self._add_dist_dir(dist.location, dist_name)
  File "/Users/x/.cache/pants/named_caches/pex_root/installed_wheels/7fea57708be4117c2d962e01abd1585a8846dc06/pex-2.1.54-py2.py3-none-any.whl/pex/pex_builder.py", line 421, in _add_dist_dir
    self._copy_or_link(path, target_dir, label=dist_name)
  File "/Users/x/.cache/pants/named_caches/pex_root/installed_wheels/7fea57708be4117c2d962e01abd1585a8846dc06/pex-2.1.54-py2.py3-none-any.whl/pex/pex_builder.py", line 541, in _copy_or_link
    self._chroot.symlink(src, dst, label)
  File "/Users/x/.cache/pants/named_caches/pex_root/installed_wheels/7fea57708be4117c2d962e01abd1585a8846dc06/pex-2.1.54-py2.py3-none-any.whl/pex/common.py", line 686, in symlink
    os.symlink(abs_src, abs_dst)
FileExistsError: [Errno 17] File exists: '/Users/x/.cache/pants/named_caches/pex_root/installed_wheels/12138e788771deb6b7f0447e95d3ff330c080b4d/PyYAML-5.3.1-cp38-cp38-macosx_11_0_x86_64.whl' -> '/private/var/folders/8j/c8jf_msj009947wyw82xvdkw0000gn/T/process-execution0ijeeU/.tmp/tmp1qqqi41k/.deps/PyYAML-5.3.1-cp38-cp38-macosx_11_0_x86_64.whl'
```

Any pointers as to what to look for etc appreciated. Will keep digging see what I can come up with..

--- a/src/python/pants/backend/python/util_rules/local_dists.py
+++ b/src/python/pants/backend/python/util_rules/local_dists.py
@@ -60,6 +60,13 @@ class LocalDistsPex:
 
     Can be consumed from another PEX, e.g., by adding to PEX_PATH.
 
+    The PEX will only contain locally built dists and not their dependencies. For Pants generated
+    `setup.py` / `pyproject.toml`, the dependencies will be included in the standard resolve process
+    that the locally-built dists PEX is adjoined to via PEX_PATH. For hand-made `setup.py` /
+    `pyproject.toml` with 3rdparty dependencies not hand-mirrored into BUILD file dependencies, this
+    will lead to issues. See https://github.com/pantsbuild/pants/issues/13587#issuecomment-974863636
+    for one way to fix this corner which is intentionally punted on for now.
+
     Lists the files provided by the dists on sys.path, so they can be subtracted from
     sources digests, to prevent the same file ending up on sys.path twice.
     """
@@ -132,6 +139,7 @@ async def build_local_dists(
             interpreter_constraints=request.interpreter_constraints,
             additional_inputs=wheels_digest,
             internal_only=request.internal_only,
+            additional_args=["--intransitive"],
         ),
     )
 

